README update with latest SDK changes and reformat (#273)
tensorlakeai/tensorlake: README.md

---

## Table of Contents

- [Features](#features)
- [Getting Started](#getting-started)
- [Document Ingestion](#document-ingestion)
  - [Document Parsing](#document-parsing)
  - [Structured Extraction](#structured-extraction)
  - [Datasets](#datasets)
- [Serverless Workflows](#serverless-workflows)
  - [Creating Workflows](#creating-workflows)
  - [Local Development](#local-development)
  - [Cloud Deployment](#cloud-deployment)
- [Additional Features](#additional-features)
  - [Webhooks](#webhooks)
  - [Batch Processing](#batch-processing)
- [Configuration Reference](#configuration-reference)
- [Examples](#examples)
- [Resources](#resources)

---

## Features

- Parse PDFs, DOCX, or PPT files into markdown
- Extract structured data using JSON Schema or Pydantic models
- Page classification, figure summarization, table extraction, signature detection
- Organize documents into auto-parsed datasets
- Deploy and run scalable workflows using a serverless cloud runtime

---

## Getting Started

### Installation

```bash
pip install tensorlake
```

### Get API Key

Sign up at [cloud.tensorlake.ai](https://cloud.tensorlake.ai/) for your API key.

## Document Ingestion

> Document Ingestion provides APIs to convert unstructured documents into structured, processable formats. This is the foundation for building RAG systems, knowledge bases, and document analysis applications.

Reviewer: ```suggestion
The Document Ingestion API converts unstructured documents into structured, processable formats. This is the foundation for building RAG systems, knowledge bases, and document analysis applications.
```

### Document Parsing

Convert documents to clean markdown or JSON while preserving layout, tables, and structure:

#### Quickstart

This uses default `ParsingOptions` to parse the document.

```python
from tensorlake.documentai import DocumentAI
from tensorlake.documentai.models import ParseStatus

doc_ai = DocumentAI(api_key="your-api-key")

# Upload and parse document
file_id = doc_ai.upload("/path/to/document.pdf")

# Get parse ID
parse_id = doc_ai.parse(file_id)

# Wait for completion and get results

> result = doc_ai.get_parsed_result(parse_id=parse_id)
> while result.status in [ParseStatus.PENDING, ParseStatus.PROCESSING]:
>     result = doc_ai.get_parsed_result(parse_id)

Reviewer: ```suggestion
result = doc_ai.wait_for_completion(parse_id)
```

```python
from tensorlake.documentai import DocumentAI
from tensorlake.documentai.models import ParsingOptions, EnrichmentOptions, ParseStatus
from tensorlake.documentai.models.enums import ChunkingStrategy, TableOutputMode

doc_ai = DocumentAI(api_key="your-api-key")

# Skip the upload step, if you are passing pre-signed URLs or HTTPS accessible files.
file_id = doc_ai.upload("/path/to/document.pdf")

# Configure parsing options
parsing_options = ParsingOptions(
    chunking_strategy=ChunkingStrategy.SECTION,
    table_output_mode=TableOutputMode.HTML,
    signature_detection=True
)

# Configure enrichment options
enrichment_options = EnrichmentOptions(
    figure_summarization=True,
    table_summarization=True
)

# Parse and wait for completion
result = doc_ai.parse_and_wait(
    file_id, 
    parsing_options=parsing_options,
    enrichment_options=enrichment_options
)

if result.status == ParseStatus.SUCCESSFUL:
    for chunk in result.chunks:
        print(chunk.content)
```

**Supported Formats:** PDF, DOCX, PPTX, images, spreadsheets, handwritten notes

**Key Features:**

> - Multiple chunking strategies (page, section, fragment)

Reviewer: ```suggestion
- Multiple chunking strategies (entire document, page, section, fragment)
```

- Table extraction and structure preservation

> - Figure and chart summarization

Reviewer: ```suggestion
- Figure and table summarization
```

#### 3. Retrieve Dataset output and metadata 
```python
items = dataset.items()
# Get dataset information
dataset_info = doc_ai.get_dataset(dataset.dataset_id)
print(f"Dataset status: {dataset_info.status}")  # idle or processing

# List all parse results for this dataset
results = doc_ai.list_parse_results(dataset_name=dataset.name)
for result in results.items:
    print(f"Parse {result.parse_id}: {result.status}")
    if result.structured_data:
        print(f"Extracted data: {result.structured_data}")

# List all datasets in your project
datasets = doc_ai.list_datasets()
for ds in datasets.items:
    print(f"Dataset: {ds.name} - Status: {ds.status}")

# Update dataset configuration
updated_dataset = doc_ai.update_dataset(
    dataset,
    description="Updated description",
    parsing_options=ParsingOptions(
        table_output_mode=TableOutputMode.MARKDOWN  # Changed table output mode
    )
)
```

A dataset can be in any of these states - `idle`, `processing`. You can also configure a webhook to receive updates for each file that is processed.

## Webhooks

You can get notified by Tensorlake when documents are ingested. Here is a [code example](examples/webhook.py).

> 

Reviewer: Let's still have a section on Webhooks - https://docs.tensorlake.ai/webhooks/overview

Reviewer 2: @drguthals The example outdated. Let's delete the example, and just point to the web hooks page in the README. Or we can just not mention web hook here anyways, since the content here is meant for copy-pastable quick start.
